src/evict/evict_lru.c

```diff
@@ -2499,18 +2527,40 @@ __wt_page_evict_urgent(WT_SESSION_IMPL *session, WT_REF *ref)
     WT_ASSERT(session, !__wt_ref_is_root(ref));
 
     page = ref->page;
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) || S2BT(session)->evict_disabled > 0)
+
+    if (S2BT(session)->evict_disabled > 0 || F_ISSET_ATOMIC_16(page, WT_PAGE_IN_URGENT_QUEUE))
         return (false);
 
-    /* Append to the urgent queue if we can. */
     cache = S2C(session)->cache;
+
+    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) && F_ISSET(cache, WT_CACHE_EVICT_ALL))
+        return (false);
+
+    /* Append to the urgent queue if we can. */
     urgent_queue = &cache->evict_queues[WT_EVICT_URGENT_QUEUE];
     queued = false;
 
     __wt_spin_lock(session, &cache->evict_queue_lock);
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) || S2BT(session)->evict_disabled > 0)
+
+    /*
+     * Check again if page eviction is disabled or the page is already in the urgent queue, as
```

Comment:
You don't have to mention the if conditions in the comment, they are self-explanatory. if you think we need the comment we can say:

```Check again, in case we raced with another thread.```

Reply:
Updated in [cc25f24](https://github.com/wiredtiger/wiredtiger/pull/8950/commits/cc25f24907ff7173c66064fd15ac99eeef1a946c)